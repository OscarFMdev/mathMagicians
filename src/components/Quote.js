import { useCallback, useState, useEffect } from 'react';
import quoteAPI from '../helpers/quoteAPI';

const Quote = () => {
  const [quote, setQuote] = useState({
    author: '',
    content: '',
  });

  const fetchData = useCallback(async () => {
    const { author, content } = await quoteAPI();
    setQuote({ author, content });
  }, []);

  useEffect(() => {
    fetchData();
  }, [fetchData]);

  return (
    <>
      <section className="quote-section">
        <p className="quote">
          {' '}
          {quote.content}
        </p>
        <p className="author">
          {quote.author}
          {' '}
        </p>
        <button type="button" onClick={fetchData}>New quote</button>
      </section>
    </>
  );
};

export default Quote;
