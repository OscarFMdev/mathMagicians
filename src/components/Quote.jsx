import { useCallback, useState, useEffect } from 'react';

const Quote = () => {
  const [quote, setQuote] = useState({
    author: '',
    content: '',
  });

  const fetchData = useCallback(() => {
    fetch('https://random-math-quote-api.herokuapp.com/')
      .then((response) => response.json())
      .then((json) => {
        setQuote(
          {
            author: json.author,
            content: json.quote,
          },
        );
      })
      .catch((error) => {
        console.log(error);
      });
  }, []);

  useEffect(() => {
    fetchData();
  }, [fetchData]);

  return (
    <>
      {
        console.log(quote)
      }
      <p className="quote">
        {' '}
        {quote.content}
      </p>
      <p className="author">
        {quote.author}
        {' '}
      </p>
    </>
  );
};

export default Quote;
