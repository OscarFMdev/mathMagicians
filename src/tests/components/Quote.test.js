import { render } from '@testing-library/react';
import Quote from '../../components/Quote';

describe('Quote', () => {
  test('should render', () => {
    const { container } = render(<Quote />);
    expect(container).toMatchSnapshot();
  });
});
