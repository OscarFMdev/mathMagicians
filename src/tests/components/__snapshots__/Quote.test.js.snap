// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote should render 1`] = `
<div>
  <section
    class="quote-section"
  >
    <p
      class="quote"
    >
       
    </p>
    <p
      class="author"
    >
       
    </p>
    <button
      type="button"
    >
      New quote
    </button>
  </section>
</div>
`;
